state/src/applicator/mod.rs

```diff
@@ -76,7 +76,7 @@ impl StateApplicator {
             },
             StateTransition::UpdateOrderMetadata { meta } => self.update_order_metadata(meta),
             StateTransition::AppendTask { task } => self.append_task(&task),
-            StateTransition::PopTask { task_id } => self.pop_task(task_id),
+            StateTransition::PopTask { task_id, success } => self.pop_task(task_id, success),
```

Comment:
this lgtm, but down the line should we just include the `QueuedTaskState` directly instead of a bool? would be more extensible if we put more states in that enum

Reply:
Yeah that's a good suggestion, i'll make that change